### PR TITLE
new image comparison pixelmatch, new image resizing jimp 

### PR DESCRIPTION
**Pull request by tnier01**
blink-diff got replaced by pixelmatch (less dependencies)
sharp got replaced by jimp (less dependencies)
tests were extended and adapted accordingly

### CODE REVIEW

**Comment by nuest** [comment]
@tnier01 Kannst du dich auch noch um die merge conflicts kümmern, bitte? Also einmal mit `upstream/master` mergen.

File: lib/checker.js [hunk 1]
```diff
@@ -219,19 +223,35 @@ function sliceImagesOutOfHTMLStringsAndCreateBuffers(readFilesArray) {
...
+				bufferedImagesOriginal.forEach(
+					function (current, index) {
+						resultingImageBuffers[index] = [current, bufferedImagesReproduced[index]]
+					}
+				)
+
+				/*
+				If there are more reproduced images than original images there is a reject.
+				*/
+				if (bufferedImagesOriginal.length < bufferedImagesReproduced.length) {
```
Comment (nuest) [inline lib/checker.js:251]: Can it only be the case the `original < reproduced` ? Why not test for `!=`?

File: lib/checker.js [hunk 2]
```diff
@@ -219,19 +223,35 @@ function sliceImagesOutOfHTMLStringsAndCreateBuffers(readFilesArray) {
...
+					reject({ error: "There are more reproduced images than original images (" + bufferedImagesOriginal.length + " originalImageBuffers and " + bufferedImagesReproduced.length + " reproducedImageBuffers)." });
```
Comment (nuest) [inline lib/checker.js:252]: Do you have a test that triggers this?

File: lib/checker.js [hunk 3]
```diff
@@ -350,124 +374,73 @@ function prepareImagesForComparison(twoDimensionalArrayOfBuffers) {
...
+				debugCompare("No resizing needed for images");
+				resolve(templateBuffer(originalImageBuffer, reproducedImageBuffer));
 			}
 		}
 	)
 }
 
 /**
- *
+ * Function which compares each of the image pairs by pixelmatch
```
Comment (nuest) [inline lib/checker.js:424]: "compares .. with pixelmatch" or "by using" ?

File: package.json [hunk 4]
```diff
@@ -1,12 +1,13 @@
 {
   "name": "erc-checker",
-  "version": "1.2.0",
+  "version": "1.2.1",
```
Comment (nuest) [inline package.json:3]: This is more than a bugfix release, so please change the version to `1.3`.

File: test/image-comparison.js [hunk 5]
```diff
@@ -23,53 +23,59 @@ const expect = require('chai').expect;
...
 
 		try {
 			fs.mkdirSync(path.join(os.tmpdir(), 'erc-checker'));
-		}catch (e) {}
-
+		} catch (e) { debug(e); }
 
 		before(function (done) {
-
+			//this.timeout(60000);
```
Comment (nuest) [inline test/image-comparison.js:49]: Please remove commented lines like this one.

File: lib/checker.js (hunk 1 above)
Comment (tnier01) [reply lib/checker.js]: good idea, changed it!

File: lib/checker.js (hunk 2 above)
Comment (tnier01) [reply lib/checker.js]: Yes there is one! https://github.com/tnier01/erc-checker/blob/master/test/testModuleUsage.js#L103

File: lib/checker.js (hunk 3 above)
Comment (tnier01) [reply lib/checker.js]: Wrote a new comment which is more clear

File: package.json (hunk 4 above)
Comment (tnier01) [reply package.json]: All right

File: test/image-comparison.js (hunk 5 above)
Comment (tnier01) [reply test/image-comparison.js]: Is removed

**Comment by nuest** [comment]
@tnier01 Changes for the changes - looking good. Now only the merge conflict must be resolved.

**Comment by tnier01** [comment]
solved problem concerning missing diff images, now everything should be fine

**Pull request merged**